rest-service/manager_rest/test/endpoints/test_app.py

```diff
@@ -29,3 +32,75 @@ def test_get_root_404(self):
         """
         resp = self.app.get('/')
         self.assertEqual(404, resp.status_code)
+
+    def test_does_not_init_tracer(self):
+        # Dummy call to invoke the tracer initialization if enabled.
+        self.app.get('/')
+        self.jaeger_mock_config.assert_not_called()
+
+
+@attr(client_min_version=1, client_max_version=base_test.LATEST_API_VERSION)
+class AppWithTracingNoIPTestCase(base_test.BaseServerTestCase):
+    """Test the basic setup of the tracer and dispatch wrapping for the
+    tracing when tracing is enabled but no endpoint IP is set.
+    """
+
+    def setUp(self):
+        super(AppWithTracingNoIPTestCase, self).setUp()
+        # Dummy call to invoke the tracer initialization if enabled.
+        self.app.get('/')
+
+    def create_configuration(self):
+        test_config = super(
+            AppWithTracingNoIPTestCase, self).create_configuration()
+        test_config.enable_tracing = True
+        test_config.tracing_endpoint_ip = None
+        return test_config
+
+    def test_does_not_init_tracer_no_tracer_ip(self):
+        self.jaeger_mock_config.assert_not_called()
+
+
+@attr(client_min_version=1, client_max_version=base_test.LATEST_API_VERSION)
+class AppWithTracingTestCase(base_test.BaseServerTestCase):
+    """Test the basic setup of the tracer and dispatch wrapping for the
+    tracing when tracing is enabled and an endpoint IP is set.
+    """
+    _tracing_endpoint_ip = 'some_ip'
+
+    def setUp(self):
+        super(AppWithTracingTestCase, self).setUp()
+        # Dummy call to invoke the tracer initialization if enabled.
+        self.app.get('/')
+
+    def create_configuration(self):
+        test_config = super(
+            AppWithTracingTestCase, self).create_configuration()
+        test_config.enable_tracing = True
+        test_config.tracing_endpoint_ip = self._tracing_endpoint_ip
+        return test_config
+
+    def test_does_init_tracer(self):
+        self.jaeger_mock_config.assert_called_with(
+            config={
+                'sampler': {'type': 'const', 'param': 1},
+                'local_agent': {
+                    'reporting_host': self._tracing_endpoint_ip},
+                'logging': True
+            },
+            service_name='cloudify-manager'
+        )
+        self.jaeger_mock_config.initialize_tracer.assert_called()
+
+    def test_adds_error_to_tags(self):
+        e = UnsupportedFormatException()
+        kwargs = {'tags': {"Extract failed": str(e)},
```

Comment:
better name is needed

Reply:
Not sure if my current solution is better. I changed it to `start_span_kwargs`. Is that better? Or not what you meant?